core/src/main/java/org/bitcoinj/core/TransactionConfidence.java

```diff
@@ -443,28 +443,43 @@ public void clearBroadcastBy() {
     }
 
     /**
-     * If this transaction has been overridden by a double spend (is dead), this call returns the overriding transaction.
+     * If this transaction has been overridden by a double spend (is dead), this call returns the overriding transaction ID.
      * Note that this call <b>can return null</b> if you have migrated an old wallet, as pre-Jan 2012 wallets did not
      * store this information.
      *
-     * @return the transaction that double spent this one
+     * @return the transaction id that double spent this one
      * @throws IllegalStateException if confidence type is not DEAD.
      */
-    public synchronized Transaction getOverridingTransaction() {
+    @Nullable
+    public synchronized Sha256Hash getOverridingTxId() {
         if (getConfidenceType() != ConfidenceType.DEAD)
             throw new IllegalStateException("Confidence type is " + getConfidenceType() +
-                                            ", not DEAD");
-        return overridingTransaction;
+                    ", not DEAD");
+        return overridingTxId;
     }
 
     /**
      * Called when the transaction becomes newly dead, that is, we learn that one of its inputs has already been spent
      * in such a way that the double-spending transaction takes precedence over this one. It will not become valid now
      * unless there is a re-org. Automatically sets the confidence type to DEAD. The overriding transaction may not
      * directly double spend this one, but could also have double spent a dependency of this tx.
+     * @deprecated Use {@link #getOverridingTxId()}
      */
+    @Deprecated
     public synchronized void setOverridingTransaction(@Nullable Transaction overridingTransaction) {
-        this.overridingTransaction = overridingTransaction;
+        Objects.requireNonNull(overridingTransaction.getTxId());
```

Comment:
Shouldn't the `requireNonNull()` be applied on the method argument, rather than the txId?

Reply:
Oops. Will fix...